app/controllers/form_controller.rb

```diff
@@ -217,7 +223,7 @@ def successful_redirect_path(pages_to_check)
       original_log = current_user.send(class_name.pluralize).find_by(id: from_referrer_query("original_log_id"))
 
       if original_log.present? && current_user.send(class_name.pluralize).duplicate_logs(original_log).any?
-        if @log.duplicates.none?
+        if @log.duplicate_set_id.nil?
```

Comment:
Assuming we trust the code that removes this if there are no longer any duplicates (which has been called before this), since we only need to know if there are any duplicates here I think we can just check if this attribute is nil - it should be nil if and only if there are no saved duplicates.